templates/chrony.conf.epp

```diff
@@ -1,12 +1,48 @@
 # NTP servers
-<% $chrony::servers.each |$server| { -%>
-server <%= $server.flatten.join(' ') %>
+<% if $chrony::servers.is_a(Hash) { -%>
```

Comment:
Does that work? I thought it would be `is_a?`. Isn't this more native Puppet syntax:
```suggestion
<% if $chrony::servers =~ Hash { -%>
```

Reply:
`is_a?` is a ruby function, `is_a` is a function from stdlib, equivalent to `=~`. I just
think the object-oriented call looks nicer, and when I se a `=~` I'm thinking regexp. But
if you want to I can change it.

Reply:
It's a bit confusing to me because I'm so used to ERB templates where it would be invalid. Not sure what the best practice would be here. Given the repetition, perhaps a private function `chrony::parameter_as_hash()` would be useful to simplify the template? https://puppet.com/docs/puppet/6.17/lang_iteration.html#using-iteration-to-transform-data suggests you can do this with pure Puppet.

Reply:
This could work.

```epp
<% $chrony::servers.each |$server| { -%>
server <%= $server.flatten.filter |$value| { $value }.join(' ') %>
<% } -%>
```

i.e. Use flatten to take care of the permitted types, and use filter to remove the
resulting empty element before joining.